spark/src/main/scala/org/apache/spark/sql/delta/commands/CreateDeltaTableCommand.scala

```diff
@@ -631,10 +632,14 @@ case class CreateDeltaTableCommand(
     operation match {
       case _ if tableByPath => // do nothing with the metastore if this is by path
       case TableCreationModes.Create =>
-        spark.sessionState.catalog.createTable(
-          cleaned,
-          ignoreIfExists = existingTableOpt.isDefined || mode == SaveMode.Ignore,
-          validateLocation = false)
+        if (createTableFunc.isDefined) {
+          createTableFunc.get.apply(cleaned)
+        } else {
+          spark.sessionState.catalog.createTable(
```

Comment:
For my understanding: Once we fix the V2SessionCatalog bug in Apache Spark, I guess we can always pass a `createTableFunc` in all cases. How do we take care of `ignoreIfExists` and `validateLocation` parameters? i.e. what is the default behavior of `super.createTable(ident, t.columns(), t.partitioning, t.properties)` w.r.t. `ignoreIfExists` and `validateLocation` parameters?

Reply:
We will need to catch `TableAlreadyExistsException` to handle `ifExists`. There is no location validation in v2 catalog `createTable` so we are fine.